File: Frontend/src/Models/Notifications/Responses/NotificationResponse.ts
export interface NotificationResponse {
    id: string;
    senderId: string;
    senderName: string;
    senderProfilePicture: string;
    type: 'NewMessage' | 'NewPost' | 'FriendRequest' | 'FriendRequestAccepted' | 'Like' | 'Comment';
    message: string;
    referenceId?: string;
    isRead: boolean;
    createdAt: string;
}

File: Backend/SocialMedia.Application/DTOs/Responses/Notifications/NotificationResponse.cs
namespace SocialMedia.Application.DTOs.Responses.Notifications;
public class NotificationResponse
{
    public Guid Id { get; set; }
    public Guid SenderId { get; set; }
    public string SenderName { get; set; } = string.Empty;
    public string SenderProfilePicture { get; set; } = string.Empty;
    public string Type { get; set; } = string.Empty;   
    public string Message { get; set; } = string.Empty;
    public Guid? ReferenceId { get; set; }
    public bool IsRead { get; set; }
    public DateTime CreatedAt { get; set; }
}
